pkg/reconciler/cloudeventingress/resources/service.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resources
+
+import (
+	"fmt"
+
+	"github.com/knative/eventing-sources/pkg/apis/sources/v1alpha1"
+	servingv1alpha1 "github.com/knative/serving/pkg/apis/serving/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// CloudEventIngressArgs are the arguments needed to create a CloudEvent Ingress. Every
+// field is required.
+type CloudEventIngressArgs struct {
+	Image   string
+	Source  *v1alpha1.CloudEventIngress
+	SinkURI string
+}
+
+// MakeCloudEventIngress generates (but does not insert into K8s) the CloudEventIngress Deployment
+// and Service for CloudEvent Sources.
+func MakeCloudEventIngress(args *CloudEventIngressArgs) *servingv1alpha1.Service {
+	labels := map[string]string{
+		"knative-eventing-source": "cloudeventingress",
+	}
+
+	return &servingv1alpha1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: fmt.Sprintf("%s-", args.Source.Name),
```

Comment:
We probably shouldn't use GenerateName. I think that people will want to have exact control over the name of the ingress' URL. Instead we should use something like `Name: fmt.Sprintf("%s-ce-ingress", args.Source.Name)`